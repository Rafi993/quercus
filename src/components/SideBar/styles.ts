import styled from 'styled-components';

export const SideBar = styled.div`
  position: sticky;
  width: 8rem;
  > div {
    position: fixed;
    top: 4.5em;
    bottom: 0;
    width: 8rem;
  }
`;
