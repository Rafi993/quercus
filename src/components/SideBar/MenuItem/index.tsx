import { connect } from 'react-redux';
import React from 'react';

import { App } from '../../../types/state';
import Icon from '../../Icon';
import { setCurrentPath } from '../../../actions/sideBar';
import { MenuItem as StyledMenuItem } from './styles';

interface Props {
  name: string;
  text: string;
  path: string;
  currentPath: string;
  _setCurrentPath: Function;
}

const MenuItem: React.FC<Props> = ({
  name,
  path,
  text,
  currentPath,
  _setCurrentPath,
}) => {
  return (
    <StyledMenuItem
      onClick={() => _setCurrentPath(path)}
      key={name}
      active={path === currentPath}
    >
      <Icon type={name} />
      <span>{text}</span>
    </StyledMenuItem>
  );
};

const mapStateToProps = (app: App) => ({
  currentPath: app.currentPath,
});

export default connect(mapStateToProps, { _setCurrentPath: setCurrentPath })(
  MenuItem
);
