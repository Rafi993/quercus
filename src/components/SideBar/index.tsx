import React from 'react';

import { defaultSideBarOptions } from '../../constants/sideBar';
import Icon from '../Icon';
import { SideBar as StyledSideBar, MenuItem } from './styles';

const SideBar: React.FC = () => {
  return (
    <StyledSideBar>
      {defaultSideBarOptions.map(item => (
        <MenuItem key={item.name}>
          <Icon type={item.name} />
          <span>{item.text}</span>
        </MenuItem>
      ))}
    </StyledSideBar>
  );
};

export default SideBar;
