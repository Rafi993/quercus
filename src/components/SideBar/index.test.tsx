import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import SideBar from './index';

afterEach(cleanup);

jest.mock('../../utils', () => {
  return {
    getTrashDir: () => 'trash',
  };
});

jest.mock('./MenuItem', () => {
  const MockMenuItem = ({ name, text }: { name: string; text: string }) => (
    <div data-name={name}>{text}</div>
  );
  return MockMenuItem;
});

it('Checking if there are all default options', () => {
  const { asFragment } = render(<SideBar />);
  expect(asFragment()).toMatchSnapshot();
});
