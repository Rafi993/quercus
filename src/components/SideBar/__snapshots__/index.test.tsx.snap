// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checking if there are all default options 1`] = `
<DocumentFragment>
  <div
    class="sc-bdVaJa vGBRp"
  >
    <div>
      <div
        data-name="recent"
      >
        Recent
      </div>
      <div
        data-name="home"
      >
        Home
      </div>
      <div
        data-name="desktop"
      >
        Desktop
      </div>
      <div
        data-name="download"
      >
        Downloads
      </div>
      <div
        data-name="image"
      >
        Pictures
      </div>
      <div
        data-name="trash"
      >
        Trash
      </div>
    </div>
  </div>
</DocumentFragment>
`;
